pkg/destroy/gcp/gcp.go

```diff
@@ -261,6 +261,16 @@ func (o *ClusterUninstaller) isClusterResource(name string) bool {
 	return strings.HasPrefix(name, o.ClusterID+"-")
 }
 
+func (o *ClusterUninstaller) isOwnedResource(labels map[string]string) bool {
```

Comment:
Based on current experience, we should also check if the private zone is an exclusively owned resource i.e it does not have other `shared` tags on it. The private zone created by cluster A could be passed in as a pre-existing private zone to cluster B, in which case, we should not be deleting the zone. 
I might have missed this, but are we adding the `owned` tag when we create the private zone and adding the `shared` tag when we are passed in a pre-existing zone?

Reply:
Right here we add the tag https://github.com/openshift/installer/blob/main/pkg/infrastructure/gcp/clusterapi/dns.go#L176.

Reply:
Do we need to explicitly find "shared" We are really looking for owned since it should be owned with this cluster ID in the key.

Reply:
Agree with BYO zone behavior, but the currently cluster could also be creating this zone which then is passed in as BYO zone to a different cluster. In that case,

- Creation: Add `owned` tag
- Deletion: Check if other `shared` tags are present, if yes, just remove `owned` tag and do not delete.
If no `shared` tags present, then actually delete the zone.